jobsdb/integration_test.go

```diff
@@ -1295,3 +1296,10 @@ func getPayloadSize(t *testing.T, jobsDB JobsDB, job *JobT) (int64, error) {
 
 	return size, err
 }
+
+func getRefreshedDSList(jobDB *HandleT) []dataSetT {
+	jobDB.dsListLock.WithLock(func(l lock.DSListLockToken) {
+		jobDB.refreshDSList(l)
+	})
+	return jobDB.getDSList()
+}
```

Comment:
I don't think making this change & every time calling `jobDB.refreshDSList()` would be helpful. Given that, in `addNewDSLoop` func we refresh the DS cache after adding new DS.

Reply:
@saurav-malani I added this because I kept getting an empty DS list in my test, this is how I solved it.

What alternative do you suggest?

The failing test was `should migrate the datasets after maxDSRetentionPeriod (except the right most one)`.

Reply:
To be more precise, I was getting a length of zero [here](https://github.com/rudderlabs/rudder-server/blob/2134c54d1b10fe40a15abeac153f15982b0509ad/jobsdb/integration_test.go#L583).

Reply:
getting this error after turning off parallel tests? Was getting the same error, but only with parallel test execution.

Reply:
Yes because the first thing I did was to replace `ginkgo` with the `go test` you see in the `Makefile`, so I was definitely getting that without parallelism. The only thing I'm doing now is shuffling, doesn't seem related though.

Reply:
Note that the go test runs the tests for different packages in parallel by default, unless the -p 1 is passed to it

Reply:
And "different packages" fine in this case is not fine? You think it should be disabled?

Reply:
no, different packages are not fine :)

Reply:
Enforcing `-p=1` for now till we got rid of global variables throughout the codebase (e.g. logger, config, etc...).